src/psyclone/transformations.py

```diff
@@ -1669,3 +1669,118 @@ def apply(self, outer):  # pylint: disable=arguments-differ
+        # single node. If the latter then we create ourselves a
+        # list containing just that node.
+        from psyclone.psyGen import Node
+        if isinstance(nodes, list) and isinstance(nodes[0], Node):
+            node_list = nodes
+        elif isinstance(nodes, Node):
+            node_list = [nodes]
+        else:
+            arg_type = str(type(nodes))
+            raise TransformationError("Error in {1}. "
+                                      "Argument must be a single Node in a "
+                                      "schedule or a list of Nodes in a "
+                                      "schedule but have been passed an "
+                                      "object of type: {0}".
+                                      format(arg_type, str(self)))
+
+        # Keep a reference to the parent of the nodes that are to be
+        # enclosed within a parallel region. Also keep the index of
+        # the first child to be enclosed as that will become the
+        # position of the new !$omp parallel directive.
+        node_parent = node_list[0].parent
+        node_position = node_list[0].position
+
+        # We need to make sure that the nodes are consecutive children,
+        # otherwise code might get moved in an incorrect order
+        prev_position = -1
+        for child in node_list:
+            if child.parent is not node_parent:
+                raise TransformationError(
+                    "Error in {0} transformation: supplied nodes "
+                    "are not children of the same Schedule/parent."
+                    .format(str(self)))
+            if prev_position>=0 and prev_position+1 != child.position:
+                raise TransformationError(
+                    "Children are not consecutive children of one parent: " 
+                    "child '{0}' has position {1}, but previous child had "
+                    "position {2}."
+                    .format(str(child), child.position, prev_position))
+            prev_position = child.position
+
+        # create a memento of the schedule and the proposed
+        # transformation
+        schedule = node_list[0].root
+
+        from psyclone.undoredo import Memento
+        keep = Memento(schedule, self)
+
+        from psyclone.profiler import ProfileNode
+        profile_node = ProfileNode(parent=node_parent, children=node_list[:])
+
+        # Change all of the affected children so that they have
+        # the ProfileNode astheir parent. Use a slice
+        # of the list of nodes so that we're looping over a local
+        # copy of the list. Otherwise things get confused when
+        # we remove children from the list.
+        for child in node_list[:]:
+            # Remove child from the parent's list of children
+            node_parent.children.remove(child)
+            child.parent = profile_node
+
+        # Add the OpenMP region directive as a child of the parent
```

Comment:
Please update this comment. If this transform has a lot in common with the OpenMP one could they share code somehow? Actually, I've just looked and I don't think there is much in the way of re-use.

Reply:
Yes - tbh, the openmp transform is broken (as far as I can see: call it with (schedule.children[2], schedule.children[1]) as parameter, and as far as I can see it will silently swap the two nodes without any test. I intend to open a separate bug (and pull request with an x-failing test), and once this pull request is merged in, refactor the common code into a new function or baseclass. But I didn't want to include this whole issue in this pull request.